web/src/components/FilingsCalendar.test.tsx

```diff
@@ -409,6 +409,62 @@ describe("<FilingsCalendar />", () => {
       expect(screen.queryByTestId("button-container")).not.toBeInTheDocument();
       process.env.FEATURE_TAX_CALENDAR = "true";
     });
+
+    it("displays button on filings calendar in up and running guest mode for non sp/gp", () => {
+      const dueDate = getCurrentDate().add(2, "months");
+      const annualReport = generateTaxFiling({
+        identifier: "annual-report",
+        dueDate: dueDate.format("YYYY-MM-DD"),
+      });
```

Comment:
just in terms of the simplest possible setup for the test, this could be removed. The existence (and due date) of the annual report doesn't impact whether the Get Access button shows or hides, so this setup is extraneous

Reply:
(and removing it will make your test much simpler :) )

Reply:
same comment for the test below

Reply:
I changed it to only contain specific userData as that's the only thing affecting the behavior.